packages/appsync-modelgen-plugin/src/utils/process-connections-v2.ts

```diff
@@ -20,19 +20,51 @@ export function getConnectedFieldV2(
     throw new Error(`The ${field.name} on model ${model.name} is not connected`);
   }
 
+  const references = connectionInfo.arguments.references;
   if (connectionInfo.name === 'belongsTo') {
     let connectedFieldsBelongsTo = getBelongsToConnectedFields(model, connectedModel);
+    if (references) {
+      if (connectionInfo) {
```

Comment:
`connectionInfo` can never be falsy since you're checking for `connectionInfo.name` on L24 above